### PR TITLE
ch-image: delete: remove multistage images

### PR DESCRIPTION
**Pull request by BenStormer**
Addresses #1368

### CODE REVIEW

File: doc/ch-image_desc.rst [hunk 1]
```diff
@@ -126,7 +126,7 @@ architecture is not available, the error message will list which ones are.
 4. Registries treat architecture as a pair of items, architecture and
    sometimes variant (e.g., "arm" and "v7"). Charliecloud treats
    architecture as a simple string and converts to/from the registry view
-   transparently.
```
Comment (j-ogas) [inline doc/ch-image_desc.rst:129]: Good catch.

File: test/build/50_ch-image.bats [hunk 2]
```diff
@@ -44,18 +44,30 @@ setup () {
     # slash present.
     ch-image build -t delete/test -f - . << 'EOF'
 FROM 00_tiny
+FROM 00_tiny
+FROM 00_tiny
+FROM 00_tiny
+FROM 00_tiny
```
Comment (j-ogas) [inline test/build/50_ch-image.bats:50]:
```suggestion
```

I think three stages is plenty for testing purposes.

File: test/build/50_ch-image.bats [hunk 3]
```diff
@@ -44,18 +44,30 @@ setup () {
...
 EOF
     run ch-image list
     echo "$output"
     [[ $status -eq 0 ]]
     [[ $output = *"delete/test"* ]]
+    [[ $output = *"delete/test_stage0"* ]]
+    [[ $output = *"delete/test_stage1"* ]]
+    [[ $output = *"delete/test_stage2"* ]]
+    [[ $output = *"delete/test_stage3"* ]]
```
Comment (j-ogas) [inline test/build/50_ch-image.bats:59]:
```suggestion
```

File: test/build/50_ch-image.bats [hunk 4]
```diff
@@ -44,18 +44,30 @@ setup () {
...
     # Delete image.
     ch-image delete delete/test
     run ch-image list
     echo "$output"
     [[ $status -eq 0 ]]
     [[ $output != *"delete/test"* ]]
+    [[ $output != *"delete/test_stage0"* ]]
+    [[ $output != *"delete/test_stage1"* ]]
+    [[ $output != *"delete/test_stage2"* ]]
+    [[ $output != *"delete/test_stage3"* ]]
```
Comment (j-ogas) [inline test/build/50_ch-image.bats:70]:
```suggestion
```

**Review by j-ogas — CHANGES_REQUESTED** [review]
Nice. See requested changes in-line.

File: test/build/50_ch-image.bats (hunk 2 above)
Comment (reidpr) [reply test/build/50_ch-image.bats]: I agree. Relevant here: The three most important numbers in computer science: zero, one, and more than one. 😉

File: doc/ch-image_desc.rst (hunk 1 above)
Comment (reidpr) [reply doc/ch-image_desc.rst:129]: Indeed, most n00bs have trouble with leaving EOL space all over.

**Review by reidpr — COMMENTED** [review]
Great work! See tidy commit and some comments in-line.

File: README.rst [hunk 5]
```diff
@@ -67,6 +67,7 @@ Contributors:
 * Megan Phinney <mphinney@lanl.gov>
 * Reid Priedhorsky <reidpr@lanl.gov>, co-founder and project lead
 * Tim Randles <trandles@lanl.gov>, co-founder
+* Benjamin Stormer <bstormer@lanl.gov>
```
[the remark this answers is not in the document]
Comment (BenStormer) [reply README.rst:70]: Great suggestion to the read-me, and I saw your commit and comments. Thanks!

**Pull request merged**